pkg/mixer/eval.go

```diff
@@ -21,16 +21,37 @@ import (
 )
 
 func evaluatePrometheusAlerts(vm *jsonnet.VM, filename string) (string, error) {
-	snippet := fmt.Sprintf("(import '%s').prometheusAlerts", filename)
+	snippet := fmt.Sprintf(`
+local mixin = (import %q);
+
+if std.objectHas(mixin, "prometheusAlerts")
+then mixin.prometheusAlerts
+else {}
+`, filename)
+
 	return vm.EvaluateSnippet("", snippet)
 }
 
 func evaluatePrometheusRules(vm *jsonnet.VM, filename string) (string, error) {
-	snippet := fmt.Sprintf("(import '%s').prometheusRules", filename)
+	snippet := fmt.Sprintf(`
+local mixin = (import %q);
+
+if std.objectHas(mixin, "prometheusRules")
+then mixin.prometheusRules
+else {}
+`, filename)
+
 	return vm.EvaluateSnippet("", snippet)
 }
 
 func evaluateGrafanaDashboards(vm *jsonnet.VM, filename string) (string, error) {
-	snippet := fmt.Sprintf("(import '%s').grafanaDashboards", filename)
+	snippet := fmt.Sprintf(`
+local mixin = (import %q);
+
+if std.objectHas(mixin, "grafanaDashboards")
+then mixin.grafanaDashboards
+else {}
+`, filename)
+
 	return vm.EvaluateSnippet("", snippet)
```

Comment:
This should use `vm.EvaluateAnonymousSnippet`, which was introduced for this exact use-case recently: https://github.com/google/go-jsonnet/pull/447

Reply:
Thats not in a release yet, and given we use go modules, we should probably wait until it is.  Please do file an issue for this.

Reply:
Done

Reply:
thank you!